src/ripple/basics/XRPAmount.h

```diff
@@ -0,0 +1,318 @@
+    , private boost::additive <XRPAmount, std::int64_t>
+{
+public:
+    using unit_type = feeunit::dropTag;
+    using value_type = std::int64_t;
+private:
+    value_type drops_;
+
+public:
+    XRPAmount () = default;
+    constexpr XRPAmount (XRPAmount const& other) = default;
+    constexpr XRPAmount& operator= (XRPAmount const& other) = default;
+
+    constexpr
+    XRPAmount (beast::Zero)
+        : drops_ (0)
+    {
+    }
+
+    constexpr
+    XRPAmount&
+    operator= (beast::Zero)
+    {
+        drops_ = 0;
+        return *this;
+    }
+
+    constexpr
+    XRPAmount (value_type drops)
+        : drops_ (drops)
+    {
+    }
+
+    XRPAmount&
+    operator= (value_type drops)
+    {
+        drops_ = drops;
+        return *this;
+    }
+
+    constexpr
+    XRPAmount
+    operator*(value_type const& rhs) const
+    {
+        return XRPAmount{ drops_ * rhs };
+    }
+
+    friend
+    constexpr
+    XRPAmount
+    operator*(value_type lhs, XRPAmount const& rhs)
+    {
+        // multiplication is commutative
+        return rhs * lhs;
+    }
+
+    XRPAmount&
+    operator+= (XRPAmount const& other)
+    {
+        drops_ += other.drops();
+        return *this;
+    }
+
+    XRPAmount&
+    operator-= (XRPAmount const& other)
+    {
+        drops_ -= other.drops();
+        return *this;
+    }
+
+    XRPAmount&
+    operator+= (value_type const& rhs)
+    {
+        drops_ += rhs;
+        return *this;
+    }
+
+    XRPAmount&
+    operator-= (value_type const& rhs)
+    {
+        drops_ -= rhs;
+        return *this;
+    }
+
+    XRPAmount&
+    operator*= (value_type const& rhs)
+    {
+        drops_ *= rhs;
+        return *this;
+    }
+
+    XRPAmount
+    operator- () const
+    {
+        return XRPAmount{ -drops_ };
+    }
+
+    bool
+    operator==(XRPAmount const& other) const
+    {
+        return drops_ == other.drops_;
+    }
+
+    bool
+    operator==(value_type other) const
+    {
+        return drops_ == other;
+    }
+
+    bool
+    operator<(XRPAmount const& other) const
+    {
+        return drops_ < other.drops_;
+    }
+
+    /** Returns true if the amount is not zero */
+    explicit
+    constexpr
+    operator bool() const noexcept
+    {
+        return drops_ != 0;
+    }
+
+    /** Return the sign of the amount */
+    constexpr
+    int
+    signum() const noexcept
+    {
+        return (drops_ < 0) ? -1 : (drops_ ? 1 : 0);
+    }
+
+    /** Returns the number of drops */
+    constexpr
+    value_type
+    drops () const
+    {
+        return drops_;
+    }
+
+    constexpr
+    double
+    decimalXRP () const;
+
+    template <class Dest>
+    boost::optional<Dest>
+    dropsAs() const
+    {
+        if ((drops_ > std::numeric_limits<Dest>::max()) ||
+            (!std::numeric_limits<Dest>::is_signed && drops_ < 0) ||
+            (std::numeric_limits<Dest>::is_signed &&
+                drops_ < std::numeric_limits<Dest>::lowest()))
```

Comment:
Fine as-is. However, we've been moving toward indenting as the `.clang_format` in the project indents. That current does not add the extra space.